src/app/header/header.component.ts

```diff
@@ -12,13 +11,55 @@ import { keyCloakService } from '../modules/admin-core';
 export class HeaderComponent implements OnInit {
   userdetails: any;
   username: any;
-  constructor(private KeycloakService: keyCloakService) { }
-  ngOnInit() {
+  user_role: any;
+  links: any[];
+  menuarray: any[];
+  constructor(private KeycloakService: keyCloakService, private usersService: UsersService) {
+
+  }
+  async ngOnInit() {
     // this.userdetails = this.KeycloakService.getKeycloakInstance();
-    this.userdetails = JSON.parse(localStorage.getItem('userdetails'));
-    console.log('**********', this.userdetails);
-    if (this.userdetails)
+    //  let res =  await this.usersService.getUserRoles();
+    this.userdetails =  JSON.parse(localStorage.getItem('userdetails'));
+     this.user_role = await localStorage.getItem('user_role');
```

Comment:
Why are you getting it from local storage? We discussed to get it from service right? If you are not using it please remove